logd/store/cursor.go

```diff
@@ -0,0 +1,35 @@
+package store
+
+import (
+	"encoding/base64"
+	"fmt"
+)
+
+var cursorEnc = base64.RawURLEncoding
+
+var NilCursor = &Cursor{
+	ID: make([]byte, 16),
+}
+
+type Cursor struct {
+	ID []byte
+}
+
+func (c Cursor) IsValid() bool {
+	return c.ID != nil
+}
+
+func ParseCursor(in string) (*Cursor, error) {
+	data, err := cursorEnc.DecodeString(in)
+	if err != nil {
+		return nil, fmt.Errorf("decoding cursor: %w", err)
```

Comment:
since this is just a pass through, safe to pass err unwrapped:

```suggestion
		return nil, err
```

general rule here is: can you add additional context? in this case, the message is basically the context of the function name, which is context that belongs to the caller.